app/services/global_search.rb

```diff
@@ -35,11 +34,28 @@ def scopes
 
   private
 
+    def validate_mode
+      return if %i[global_search searchable_association].include?(mode)
+
+      raise "Invalid mode: #{mode}"
+    end
+
     def hide_scope?(scope)
       scope[:hide_global_nav] || no_records?(scope)
     end
 
     def no_records?(scope)
       Pundit.policy_scope!(current_user, scope[:model].constantize).none?
     end
+
+    def policy_ok?(item)
```

Comment:
here's the key change, if it's a global search, it needs both index and show, but if it's a searchable association, it just needs index